js_modules/dagit/packages/app/src/NUX/CommunityNux.tsx

```diff
@@ -0,0 +1,145 @@
+  Heading,
+  Icon,
+  TextInput,
+} from '@dagster-io/ui';
+import React from 'react';
+import isEmail from 'validator/lib/isEmail';
+
+export const CommunityNux = () => {
+  const [didDismissCommunityNux, dismissCommunityNux] = useStateWithStorage(
+    'communityNux',
+    (data) => data,
+  );
+  if (didDismissCommunityNux) {
+    return null;
+  }
+  return (
+    <CommunityNuxImpl
+      dismiss={() => {
+        dismissCommunityNux('1');
+      }}
+    />
+  );
+};
+
+const CommunityNuxImpl: React.FC<{dismiss: () => void}> = ({dismiss}) => {
+  const [shouldShowNux, setShouldShowNux] = React.useState(false);
+  React.useEffect(() => {
+    const timeout = setTimeout(() => {
+      setShouldShowNux(true);
+    }, 5000); // wait 5 seconds before showing Nux
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, []);
+
+  const [email, setEmail] = React.useState('');
+  const [newsletter, setNewsLetter] = React.useState(false);
+
+  const submit = async () => {
+    await fetch('http://dagster.cloud/email-sign-up', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email,
+        newsletter,
+      }),
+    });
+    dismiss();
+  };
+
+  const validEmail = isEmail(email);
+  const [emailChanged, setEmailChanged] = React.useState(false);
+
+  return (
+    <Dialog isOpen={shouldShowNux} style={{width: '680px'}}>
+      <Box flex={{direction: 'column', gap: 16}} style={{padding: '36px'}}>
+        <Box
+          flex={{direction: 'row', gap: 24, alignItems: 'center'}}
+          padding={{bottom: 24}}
+          border={{side: 'bottom', color: Colors.KeylineGray, width: 1}}
+        >
+          <Box flex={{direction: 'column', gap: 8}}>
+            <Heading>Join the Dagster community</Heading>
+            <Body style={{color: Colors.Gray700}}>
+              Connect with thousands of other data practitioners building with Dagster. Share
+              knowledge, get help, and contribute to the open-source project.
+            </Body>
+            <div>
+              <ExternalAnchorButton
+                icon={<Icon name="slack" />}
+                href="https://www.dagster.io/slack"
+              >
+                Join us on Slack
+              </ExternalAnchorButton>
+            </div>
+          </Box>
+          <video autoPlay muted loop playsInline width={120} height={120}>
+            <source src="/Dagster_world.mp4" type="video/mp4" />
+          </video>
+        </Box>
+        <Box flex={{direction: 'column', justifyContent: 'stretch', gap: 12}}>
+          <div>Notify me about Dagster security updates</div>
+          <TextInput
+            value={email}
+            onChange={(e) => {
+              setEmail(e.target.value);
+              setEmailChanged(true);
+            }}
+            placeholder="hello@dagster.io"
+            strokeColor={!emailChanged || validEmail ? undefined : Colors.Red500}
```

Comment:
Am I understanding this correctly that while I'm typing my email address, it will appear to be in an error state until it is valid? If so, can we wait until submission to show the error state? It always feels a little weird to have an error state on something like this when I'm still typing.

Reply:
Yeah that's the correct behavior. I prefer this tbh, I think its a worse experience to click submit and then find out you made an error.